libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -425,7 +425,7 @@ static IotMqttError_t _deserializePublish( _mqttConnection_t * pMqttConnection,
     if( status != IOT_MQTT_SUCCESS )
     {
         /* Check ownership of the received MQTT packet. */
-        if( pOperation->u.publish.pReceivedData != NULL )
+        if( ( pOperation != NULL ) && ( pOperation->u.publish.pReceivedData != NULL ) )
```

Comment:
This does not appear to be sufficient. Could you change line 425 to skip this block if status indicates an allocation failure?

Reply:
+1. Line 438 also dereferences `pOperation`. It appears that `status` can only be `IOT_MQTT_NO_MEMORY` if the malloc failed, so that check might be sufficient.

Reply:
This block of code will not execute if pOperation is NULL. On line 364 we set the status to IOT_MQTT_NO_MEMORY, so the status will not be IOT_MQTT_SUCCESS. Once we set the status to something other than IOT_MQTT_SUCCESS, it should not enter any of the other blocks.

Reply:
Right I see now.  But Misra also likes this change. Maybe I can just add a comment or we can keep it as a defensive?

Reply:
Just to add to above comment for the context - this MISRA violation got introduced when I removed 'goto' after malloc failure.

Reply:
I take back my comment, the failure happens when it not IOT_MQTT_SUCCESS. Original cleanup label was pointing directly above return. Thanks Muneeb.

Reply:
Let me go back and fix this more carefully.